pycrdt_websocket/websocket_server.py

```diff
@@ -198,11 +209,20 @@ async def start(
             if self._task_group is not None:
                 raise RuntimeError("WebsocketServer already running")
 
-            async with create_task_group() as self._task_group:
-                task_status.started()
-                self.started.set()
-                # wait forever
-                self._task_group.start_soon(Event().wait)
+            while True:
+                try:
+                    async with create_task_group() as self._task_group:
+                        if not self.started.is_set():
+                            task_status.started()
+                            self.started.set()
+                        # wait forever
+                        self._task_group.start_soon(Event().wait)
+                    break
+                except Exception as e:
```

Comment:
When the original task group fail, all its tasks are cancelled, including all the existing rooms's broadcast_update tasks, do we need to find a way to recover them? 
Is it possible to have one task group per room so one room throws exception in broadcast_update task won't also tear down other room's  broadcast_update task?

Reply:
Could we also add auto_restart for yroom's task group as well?

Reply:
I think we should explore recovering each individual room too, but maybe let's leave this for a follow-up PR? 

At the _very minimum_ we need recovery at the top layer.